Implemented WaterSwitch functionalty

ThorLie opened the pull request:
Implemented WaterSwitch functionalty by introducing a waterHandler.h following the sample of steamHandler.h for the switch of the machinestate. Introduced new machineState kWater. The functionality is also handled in the waterHandler.h. Created editableVars and mqttVars.
Tested hardware setup on the bench. Tested MQTT (waterOn status is sent).

ThorLie committed:
First Adjustment WaterSwitch: Created Software Switch and created handler file.

ThorLie committed:
Implemented WaterSwitch functionalty by introducing a waterHandler.h following the sample of steamHandler.h for the switch of the machinestate. Introduced new machineState kWater. The functionality is also handled in the waterHandler.h. Created editableVars and mqttVars.
Tested hardware setup on the bench. Test MQTT (waterOn status is sent).

ThorLie committed:
Removed annotations

ThorLie committed:
removed WaterFirstOn since it is not used or implemented.

ThorLie committed:
deleted waterOn functionality outside of waterHandler.h

ThorLie committed:
Deleted MQTT functionality of waterOn

ThorLie committed:
added FEATURE-WATERSWITCH to userconfig_sample

fiendie commented on src/waterHandler.h:
```diff
@@ -0,0 +1,45 @@
+        uint8_t waterSwitchReading = waterSwitch->isPressed();
+
+        if (WATERSWITCH_TYPE == Switch::TOGGLE) {
+            // Set waterON to 1 when steamswitch is HIGH
```
Should be water switch, probably a copy-paste issue

fiendie commented on src/waterHandler.h:
```diff
@@ -0,0 +1,45 @@
+                pumpRelay.on();
+            }
+
+            // if activated via web interface then steamFirstON == 1, prevent override
```
See above, copy-pasted from steamHandler.h

fiendie commented on src/main.cpp:
```diff
@@ -997,7 +1056,9 @@ void handleMachineState() {
-                else if (isBrewDetected) {
+                else if (waterON) {
+                    machineState = kWater;
+                }else if (isBrewDetected) {
```
Missing line break, please run clang-format.

ThorLie committed:
Implemented corrections (copy-paste errors, format, default value)

ThorLie commented on src/waterHandler.h:
```diff
@@ -0,0 +1,45 @@
+            if (waterSwitchReading != currStateWaterSwitch) {
+                currStateWaterSwitch = waterSwitchReading;
+
+                // only toggle heating power if the new button state is HIGH
```
Thanks @fiendie for the remarks. Highly appeciated.

LoQue90 committed:
reorder machineState for kwater

LoQue90 committed:
rename waterswitch related vars

to be more presice for waterswitch and separating from water tank:
kWater ->kHotWater
waterON -> hotWaterOn

ThorLie commented:
@LoQue90 renaming water to hot water is a good idea.

LoQue90 committed:
add hot water screen

showing water drops and current temperature

LoQue90 committed:
rename Feature, Switch and Pin

be consistend with naming:
FEATURE_HOTWATERSWITCH
PIN_HOTWATERSHWITCH
hotWaterHandler
hotWaterSwitch

LoQue90 committed:
do not allow pulling hot water if water tank is empty

moj02090 commented:
@ThorLie  thanks for providing the feature pr. I've changed the wiring the past days and today I was able to test the PR in my Rancilio Silvia V4 @ Clevercoffee PCB Rev 1.6 without any issues yet. If I find any problems I will let you know.

LoQue90 committed:
do not allow backflush while hot water is running

LoQue90 committed:
pulling hotWater is only possible from kPidNormal and KSteam

LoQue90 committed:
Update main.cpp

LoQue90 commented:
added with #546
